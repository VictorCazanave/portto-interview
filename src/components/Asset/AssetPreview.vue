<template>
	<div class="preview">
		<img
			:src="asset.imageUrl"
			:alt="asset.name"
			class="image"
		>

		<span class="name">
			{{ asset.name }}
		</span>
	</div>
</template>

<script>
export default {
	name: 'AssetPreview',
	props: {
		asset: {
			type: Object,
			required: true,
		},
	},
}
</script>

<style lang="scss" scoped>
.preview {
	display: flex;
	flex-direction: column;
	align-items: center;
	border: 2px solid $color-black;
	padding: 1rem;

	&:focus,
	&:hover {
		background: $color-grey;
	}
}

.image {
	height: 10rem; // NOTE: Arbitrary value
	width: auto;
	margin-bottom: 1rem;
}

.name {
	font-size: 1.2rem;
}
</style>
