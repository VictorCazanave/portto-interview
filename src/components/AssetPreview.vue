<template>
	<div>
		<img
			:src="asset.imageUrl"
			:alt="asset.name"
		>
		<span>
			{{ asset.name }}
		</span>
	</div>
</template>

<script>
export default {
	name: 'AssetPreview',
	props: {
		asset: {
			type: Object,
			required: true,
		},
	},
}
</script>

<style lang="scss">
</style>
