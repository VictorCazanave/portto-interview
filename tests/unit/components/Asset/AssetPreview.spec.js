import { shallowMount } from '@vue/test-utils'
import AssetPreview from '@/components/Asset/AssetPreview.vue'

describe('AssetPreview components', () => {
	const asset = {
		name: 'name',
		imageUrl: 'imageUrl',
	}

	it('renders asset data properly', () => {
		const wrapper = shallowMount(AssetPreview, {
			propsData: { asset },
		})

		expect(wrapper.element).toMatchSnapshot()
	})
})
