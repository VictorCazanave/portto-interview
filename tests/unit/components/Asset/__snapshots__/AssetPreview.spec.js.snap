// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetPreview components renders asset data properly 1`] = `
<div
  class="preview"
>
  <img
    alt="name"
    class="image"
    src="imageUrl"
  />
   
  <span
    class="name"
  >
    
		name
	
  </span>
</div>
`;
